crates/modelardb_server/src/storage/compressed_data_manager.rs

```diff
@@ -162,13 +162,16 @@ impl CompressedDataManager {
         };
 
         // Update the remaining memory for compressed data and record the change.
-        self.compressed_remaining_memory_in_bytes -= segments_size as isize;
         self.used_compressed_memory_metric
             .append(segments_size as isize, true);
 
         // If the reserved memory limit is exceeded, save compressed data to disk.
-        if self.compressed_remaining_memory_in_bytes < 0 {
-            self.save_compressed_data_to_free_memory().await?;
+        if self
```

Comment:
Should this be negated since it returns [`true`] if the reservation succeeds which means we would want to free memory if the reservation did not succeed.

Reply:
Furthermore, if `try_reserve_compressed_memory` returns `false` i am not sure we ever retry to reserve the compressed memory after compressed data is saved. I think this was handled before by going to less than 0 before the check and then just saving until we were above 0 again.

Reply:
Fixed by changing it to `while ! ...`.